### PR TITLE
(update) remove old settings

### PR DESCRIPTION
**Pull request by ilyaDymshits**

### CODE REVIEW

File: amd/src/annoto.js [hunk 1]
```diff
@@ -427,90 +376,79 @@ define([
...
-                    position: params.position,
-                    features: {
-                        tabs: params.featureTab,
-                        cta: params.featureCTA,
-                    },
-                    align: {
-                        vertical: params.alignVertical,
-                    },
                     ux: {
                         ssoAuthRequestHandle: function() {
```
Comment (genadis) [inline amd/src/annoto.js:418]: `ssoAuthRequestHandle` should be moved to hooks

File: amd/src/annoto.js [hunk 2]
```diff
@@ -427,90 +376,79 @@ define([
...
+                                    id: params.mediaId,
+                                    description: params.$cmintro,
+                                }
+                            };
                         },
-                    }],
-                    demoMode: params.demoMode,
+                    },
                     locale: params.locale,
+                    widgets: [{player: {}}],
```
Comment (genadis) [inline amd/src/annoto.js:434]: remove `widgets: [{player: {}}],`

File: amd/src/annoto.js [hunk 3]
```diff
@@ -427,90 +376,79 @@ define([
...
-                                    description: params.mediaDescription,
-                                    group: {
-                                        id: params.mediaGroupId,
-                                        type: 'playlist',
-                                        title: params.mediaGroupTitle,
-                                        privateThread: params.privateThread,
-                                    }
-                                };
-                            },
+                                    id: params.mediaId,
```
Comment (genadis) [inline amd/src/annoto.js:427]: as in other comment, will break exiting customer's discussions

File: amd/src/annoto.js [hunk 4]
```diff
@@ -427,90 +376,79 @@ define([
...
+                                    description: params.$cmintro,
```
Comment (genadis) [inline amd/src/annoto.js:428]: shouldn't this be `arams.mediaDescription` ?

File: amd/src/annoto.js [hunk 5]
```diff
@@ -427,90 +376,79 @@ define([
...
+                        description: params.mediaDescription,
+                    },
                 });
             });
 
-            annoto.onReady(function (api) {
-                // recomended, so notifications will have URL to valid pages
+            annoto.onReady(function(api) {
+                // Recomended, so notifications will have URL to valid pages
                 api.registerOriginProvider({
```
Comment (genadis) [inline amd/src/annoto.js:445]: no such api.
`getPageUrl` should be moved to hooks

**Review by genadis — CHANGES_REQUESTED** [review]
Please fix the review comments.

In addition, shouldn't we have settings for the new dashboard capability as well?

image

File: amd/src/annoto.js [hunk 6]
```diff
@@ -300,32 +271,13 @@ define([
...
-            config.position = params.position;
-            config.demoMode = params.demoMode;
             config.locale = params.locale;
-            config.rtl = params.rtl;
-
-            features.tabs = params.featureTab;
-            features.cta = params.featureCTA;
-            align.vertical = params.alignVertical;
-            ux = {
-                ssoAuthRequestHandle: function() {
```
[the remark this answers is not in the document]
Comment (genadis) [reply amd/src/annoto.js:324]: In addition, please add `getPageUrl` hook to kaltura as well similar to wistia

File: lib.php [hunk 7]
```diff
@@ -357,15 +324,6 @@ function local_annoto_create_lti_course_module(){
...
 
-    // Get plugin global settings.
-    $settings = get_config('local_annoto');
-    $lti = lti_get_tool_by_url_match($settings->toolurl);
-
-    if (!$lti){
-        $lti = new stdClass();
-        $lti->id = local_annoto_lti_add_type();
-    }
-
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply lib.php:368]: Should be same

File: db/access.php [hunk 8]
```diff
@@ -36,4 +36,17 @@
         ),
         'clonepermissionsfrom' => 'moodle/course:manageactivities'
     ),
+
+    'local/annoto:managementdashboard' => array(
+        'riskbitmask' => RISK_SPAM,
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply db/access.php]: ready for checks

File: lang/en/local_annoto.php [hunk 9]
```diff
@@ -46,7 +44,6 @@
 $string['customdomaindesc'] = 'Specify a custom deployment domain. Please note that the widget data is bound to a specific region.';
 $string['custom'] = 'Custom';
 
-
 // Application settings.
 $string['appsetingsheading'] = 'Annoto settings';
 $string['cta'] = 'Call to action';
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply lang/en/local_annoto.php]: ready for checks

File: lang/en/local_annoto.php [hunk 10]
```diff
@@ -56,68 +53,42 @@
...
-$string['discussionscope'] = 'Discussions scope';
-$string['discussionscopedesc'] = 'Choose scope of Annoto discussions: private per course (default) or site wide (if same video is used in multiple courses the discussion will be pulic across courses)';
-$string['discussionscopesitewide'] = 'Site wide';
-$string['discussionscopeprivate'] = 'Private per course';
 $string['moderatorroles'] = 'Moderator roles';
 $string['moderatorrolesdesc'] = 'Specify who is allowed to moderate discussions (only roles that at least include the following capabilities are available: local/annoto:moderatediscussion).';
+$string['addingdashboard'] = 'Add Annoto dashboard';
+$string['addingdashboard_desc'] = 'Add Annoto dashboard automatically to each course';
 
 // UX preferences.
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply lang/en/local_annoto.php]: ready for checks

File: lang/en/local_annoto.php [hunk 11]
```diff
@@ -56,68 +53,42 @@
 $string['localeauto'] = 'Auto detect';
 $string['localeen'] = 'English';
 $string['localehe'] = 'Hebrew';
-$string['discussionscope'] = 'Discussions scope';
-$string['discussionscopedesc'] = 'Choose scope of Annoto discussions: private per course (default) or site wide (if same video is used in multiple courses the discussion will be pulic across courses)';
-$string['discussionscopesitewide'] = 'Site wide';
-$string['discussionscopeprivate'] = 'Private per course';
 $string['moderatorroles'] = 'Moderator roles';
 $string['moderatorrolesdesc'] = 'Specify who is allowed to moderate discussions (only roles that at least include the following capabilities are available: local/annoto:moderatediscussion).';
+$string['addingdashboard'] = 'Add Annoto dashboard';
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply lang/en/local_annoto.php]: ready for checks

File: lang/en/local_annoto.php [hunk 12]
```diff
@@ -56,68 +53,42 @@
...
-
-// Privacy API.
-$string['privacy:metadata:annoto'] = 'In order to integrate with a remote service, user data needs to be exchanged with that service.';
-$string['privacy:metadata:annoto:userid'] = 'The userid is sent from Moodle to allow you to access your data on the remote system.';
-$string['privacy:metadata:annoto:fullname'] = 'Your full name is sent to the remote system to allow a better user experience.';
-$string['privacy:metadata:annoto:email'] = 'Your e-mail name is sent to the remote system to allow a better user experience.';
 
 // LTI
 $string['externaltoolsettings'] = 'Annoto dashboard (LTI)';
 $string['toolurl'] = 'External tool URL';
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply lang/en/local_annoto.php]: ready for checks

File: lang/en/local_annoto.php [hunk 13]
```diff
@@ -56,68 +53,42 @@
...
 $string['toolurldesc'] = 'Provide external tool URL';
 $string['tooliconurl'] = 'External tool icon';
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply lang/en/local_annoto.php]: ready for checks

File: lang/en/local_annoto.php (hunk 12 above)
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply lang/en/local_annoto.php]: ready for checks

File: lang/en/local_annoto.php [hunk 14]
```diff
@@ -56,68 +53,42 @@
...
 $string['tooliconurldesc'] = 'External tool icon URL';
-
 $string['toolname'] = 'External tool name';
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply lang/en/local_annoto.php]: ready for checks

File: amd/src/annoto.js [hunk 15]
```diff
@@ -90,127 +83,105 @@ define([
...
-                horizontalAlign = 'inner';
-            }
+                nonOverlayTimelinePlayers = ['youtube', 'vimeo'];
 
-            config.width = {max: (horizontalAlign === 'inner') ? 320 : 360};
-            config.align.horizontal = horizontalAlign;
             config.widgets[0].player.type = params.playerType;
-            config.widgets[0].player.element = params.playerId;
+            config.widgets[0].player.element = params.element;
             config.timeline = {
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply amd/src/annoto.js]: ready for checks

File: amd/src/annoto.js [hunk 16]
```diff
@@ -90,127 +83,105 @@ define([
...
-                features: {
-                    tabs: params.featureTab,
-                    cta: params.featureCTA,
-                },
-                width: {},
-                align: {
-                    vertical: params.alignVertical,
-                },
                 ux: {
                     ssoAuthRequestHandle: function() {
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply amd/src/annoto.js]: ready for checks

File: amd/src/annoto.js [hunk 17]
```diff
@@ -90,127 +83,105 @@ define([
...
-                                group: {
-                                    id: params.mediaGroupId,
-                                    type: 'playlist',
-                                    title: params.mediaGroupTitle,
-                                    privateThread: params.privateThread,
-                                }
-                            };
-                        },
+                                id: params.mediaId,
+                                description: params.$cmintro,
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply amd/src/annoto.js]: changed it for params.mediaDescription

File: amd/src/annoto.js [hunk 18]
```diff
@@ -300,32 +271,13 @@ define([
...
-            features.cta = params.featureCTA;
-            align.vertical = params.alignVertical;
-            ux = {
-                ssoAuthRequestHandle: function() {
-                    window.location.replace(params.loginUrl);
-                },
-                openOnLoad: params.openOnLoad,
-            };
+
             playerConfig.mediaDetails = this.enrichMediaDetails.bind(this);
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply amd/src/annoto.js]: ready for checks, not  sure for 100% about the solution

File: amd/src/annoto.js (hunk 2 above)
Comment (ilyaDymshits) [reply amd/src/annoto.js]: moved it to str. 149 i need it for updating  config before each widget reloading.

File: amd/src/annoto.js (hunk 3 above)
Comment (ilyaDymshits) [reply amd/src/annoto.js:425]: ready for testing

File: amd/src/annoto.js (hunk 4 above)
Comment (ilyaDymshits) [reply amd/src/annoto.js]: ready for testing

File: amd/src/annoto.js [hunk 19]
```diff
@@ -159,28 +160,25 @@ define([
...
+                                description: params.mediaDescription,
                             }
                         };
                     },
+                    ssoAuthRequestHandle: function() {
+                        window.location.replace(params.loginUrl);
+                    },
                 },
                 locale: params.locale,
-                widgets: [{player: {}}],
```
Comment (genadis) [inline amd/src/annoto.js]: why removed? it's needed here.

this comment referred to an entirely different object which is used for wistia, there it's not needed.
image

File: amd/src/annoto.js [hunk 20]
```diff
@@ -271,14 +269,25 @@ define([
...
+                id: params.mediaGroupId,
+                title: params.mediaGroupTitle,
+                description: params.mediaGroupDescription,
+            };
             config.locale = params.locale;
-
-            playerConfig.mediaDetails = this.enrichMediaDetails.bind(this);
         },
 
         enrichMediaDetails: function(details) {
```
[the remark this answers is not in the document]
Comment (genadis) [reply amd/src/annoto.js:293]: revert the logic of updating the media details to how it was, and fix according to my previous comment

**Review by genadis — CHANGES_REQUESTED** [review]

File: amd/src/annoto.js [hunk 21]
```diff
@@ -90,127 +83,105 @@ define([
...
-                                description: params.mediaDescription,
-                                group: {
-                                    id: params.mediaGroupId,
-                                    type: 'playlist',
-                                    title: params.mediaGroupTitle,
-                                    privateThread: params.privateThread,
-                                }
-                            };
-                        },
+                                id: params.mediaId,
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply amd/src/annoto.js:168]: ok , done

File: lib.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (HenEytan) [reply lib.php:368]: It's the same

File: lang/en/local_annoto.php [hunk 22]
```diff
@@ -22,102 +22,54 @@
...
-$string['usregion'] = 'US region';
-
 $string['customdomain'] = 'Custom deployment domain';
 $string['customdomaindesc'] = 'Specify a custom deployment domain. Please note that the widget data is bound to a specific region.';
+$string['eurregion'] = 'EU region';
+$string['usregion'] = 'US region';
 $string['custom'] = 'Custom';
 
+// Annoto dashboard (LTI)
+$string['addingdashboard'] = 'Add to the all courses';
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply lang/en/local_annoto.php]: Ready

File: db/access.php [hunk 23]
```diff
@@ -36,4 +36,17 @@
         ),
         'clonepermissionsfrom' => 'moodle/course:manageactivities'
     ),
+
+    'local/annoto:managementdashboard' => array(
+        'riskbitmask' => RISK_PERSONAL,
```
[the remark this answers is not in the document]
Comment (ilyaDymshits) [reply db/access.php]: ready

File: amd/src/annoto.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (genadis) [reply amd/src/annoto.js:324]: @ilyaDymshits `ssoAuthRequestHandle` is still missing in hooks, please add

**Comment by genadis** [comment]
@ilyaDymshits final comment to resolve: https://github.com/Annoto/moodle-local_annoto/pull/57#discussion_r810527786

**Pull request closed without merging**